app/Jobs/PublishTweet.php

```diff
@@ -37,10 +39,17 @@ public function handle()
         //...
         $tweet = substr($this->report->title ?? $this->report->description, 0, 150).'.. ';
         $link = route('violence-reports.show', ['violence_report' => $this->report->id ]);
-        (new TwitterService)->tweet(
-            $tweet.$link,
-            !is_null($this->report->file) ? Storage::disk('s3')->get($this->report->file) : null
-        );
-        return;
+        $CrawlerDetect = new CrawlerDetect;
+        $countryName = Location::get($this->report->ip_address)->countryName;
+
+        if($this->report->user_agent == '' || $CrawlerDetect->isCrawler($this->report->user_agent) || $countryName != 'Nigeria' || $this->report->ip_address == '') {
```

Comment:
@offiahchiaz `$this->report->ip_address` would be `NULL` if it doesn't exist, not an empty string `''`